packages/ember-template-compiler/lib/plugins/transform-link-to.ts

```diff
@@ -1,43 +1,38 @@
 import { AST, ASTPlugin, ASTPluginEnvironment } from '@glimmer/syntax';
 import { Builders } from '../types';
 
-function buildProgram(b: Builders, content: AST.Node, loc: AST.SourceLocation) {
-  return b.program([buildStatement(b, content, loc)], undefined, loc);
+function buildProgram(b: Builders, content: AST.Node, escaped: boolean, loc: AST.SourceLocation) {
+  return b.program([buildStatement(b, content, escaped, loc)], undefined, loc);
 }
 
-function buildStatement(b: Builders, content: AST.Node, loc: AST.SourceLocation) {
+function buildStatement(b: Builders, content: AST.Node, escaped: boolean, loc: AST.SourceLocation) {
   switch (content.type) {
     case 'PathExpression':
-      return b.mustache(content, undefined, undefined, undefined, loc);
+      return b.mustache(content, undefined, undefined, !escaped, loc);
 
     case 'SubExpression':
-      return b.mustache(content.path, content.params, content.hash, undefined, loc);
+      return b.mustache(content.path, content.params, content.hash, !escaped, loc);
 
     // The default case handles literals.
     default:
       return b.text(`${(content as AST.Literal).value}`, loc);
   }
 }
 
-function unsafeHtml(b: Builders, expr: AST.Expression) {
-  return b.sexpr('-html-safe', [expr]);
```

Comment:
I'm not sure why we were doing this before. `{{{link-to foo ...}}}` -> `{{#link-to ...}}{{{foo}}}{{/link-to}}` seems easier and closer in spirit to what was written